week03_model_free/display_server_preparation.sh

```diff
@@ -0,0 +1,47 @@
+echo "apt-get updating..."
+apt-get -qq update > /dev/null
+echo "x11vnc installing if it is not exists..."
+echo ""
+apt-get -q install -y x11vnc
+echo ""
+echo "x11vnc is installed"
+checkXvfbProcess=1 # 0 == True, 1 == False
+if [ "$checkXvfbProcess" -eq 0 ];
+then
+    if [ -e ./xvfb.pid ];
+    then
+        xpid="$(cat ./xvfb.pid)"
+        com="$(ps -p ${xpid} -o comm=)"
+        idealcom="Xvfb"
+        if [ "$com" != "$idealcom" ];
+        then
+            echo "Run the XVFB on display :1 (usually it means that the first cell of the notebook should be run) (2)"
+            exit 1
+        fi
+    else
+        echo "Run the XVFB on display :1 (usually it means that the first cell of the notebook should be run) (1)"
+        exit 1
+    fi
+fi
+xdpyinfo -display :1 >/dev/null 2>/dev/null
+if [ "$?" -ne 0 ]
```

Comment:
Why go through all this trouble instead of using the approach from the notebook's first cell (checking `$DISPLAY`)?

Reply:
See the _never_ satisfied condition case reply of mine